### DISCUSSION TITLE
[HELP] How to specify a fixed size for the application window? (.NET MAUI Blazor WinUI)

### DISCUSSION THREAD

**Discussion by jwiggenhorn** [dotnet/maui#2370] [Q&A] [answered]
Hi, I'm working on an application using the .NET MAUI Blazor template. This application is mainly targeting Windows and I'd like to be able to specify a fixed application window size. Is this currently possible? 

I'm able to set the size of the BlazorWebView within MainPage.xaml using the WidthRequest and HeightRequest properties but this does not affect the overall window size.

**Comment by jwiggenhorn** [comment, marked as answer]
For anyone else having this issue, here's what ended up working for me:

```
// App.xaml.cs

	public App()
	{
		InitializeComponent();

                Microsoft.Maui.Handlers.WindowHandler.WindowMapper[nameof(IWindow)] = (handler, view) =>
                {
#if WINDOWS
                	var nativeWindow = handler.NativeView;
            		nativeWindow.Activate();
            		IntPtr windowHandle = PInvoke.User32.GetActiveWindow();

            		PInvoke.User32.SetWindowPos(windowHandle, 
					PInvoke.User32.SpecialWindowHandles.HWND_TOP,
                                        0, 0, width, height,  // width and height are ints
                                        PInvoke.User32.SetWindowPosFlags.SWP_NOMOVE);
#endif
        	};
    	}
```

**Reply by Crauzer** [reply]
I refuse to believe there's no better solution that this, because having to call a native API is fairly ugly for such a simple thing.

**Reply by jwiggenhorn** [reply]
@Crauzer Yeah if you've got something better I'd love to see it.

**Reply by PaulChek** [reply]
not working in preview 12

**Reply by kla50** [reply]
The following is working for me in Preview 12.  Added this code to the MauiProgram.CreateMauiApp method.  "builder" is the MauiAppBuilder instance and "appSettings" is my settings implementation.  This restores the app window to the position and size it was at when it was closed in the previous session (works across multiple monitors).

#if WINDOWS
		builder.ConfigureLifecycleEvents(events => {
			events.AddWindows(wndLifeCycleBuilder => {
				wndLifeCycleBuilder.OnWindowCreated(window => {
					IntPtr hwnd = WinRT.Interop.WindowNative.GetWindowHandle(window);
					PInvoke.User32.SetWindowPos(hwnd, PInvoke.User32.SpecialWindowHandles.HWND_TOP,
						(Int32)appSettings.WindowsSettings.AppWindow.X, (Int32)appSettings.WindowsSettings.AppWindow.Y, (Int32)appSettings.WindowsSettings.AppWindow.Width, (Int32)appSettings.WindowsSettings.AppWindow.Height,
						PInvoke.User32.SetWindowPosFlags.SWP_SHOWWINDOW);
				});
				wndLifeCycleBuilder.OnClosed((window, args) => {
					IntPtr hwnd = WinRT.Interop.WindowNative.GetWindowHandle(window);
					if(PInvoke.User32.GetWindowRect(hwnd, out PInvoke.RECT rect1)) {
						appSettings.WindowsSettings.AppWindow.X = rect1.left;
						appSettings.WindowsSettings.AppWindow.Y = rect1.top;
						appSettings.WindowsSettings.AppWindow.Width = rect1.right - rect1.left;
						appSettings.WindowsSettings.AppWindow.Height = rect1.bottom - rect1.top;
					}
					appSettings.SaveSettings();
				});
			});
		});
#endif

**Reply by Xyncgas** [reply]
Would be nice if we can call MAUI API instead of platform specific
Unlike maintaining a bunch of components, it's basic layout, I understand not every platform let you adjust windows size but you can just do nothing if it's not supported (no exception, return void, add separate method for checking whether the platform supports specifying window sizes)

**Comment by nor0x** [comment]
I have another option that doesn't rely on PInvoke directly and works >preview12. To work with `AppWindow` from `Microsoft.UI.Windowing` you need to get the `WindowId` from the window handle.

The `AppWindow` object provides functionality like `Resize `or `MoveAndResize`



```
        Microsoft.Maui.Handlers.WindowHandler.Mapper.AppendToMapping(nameof(IWindow), (handler, view) =>
        {
#if WINDOWS
            var nativeWindow = handler.PlatformView;
            nativeWindow.Activate();
            IntPtr hWnd = WinRT.Interop.WindowNative.GetWindowHandle(nativeWindow);
            WindowId windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(hWnd);
            AppWindow appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);
            //TODO: this is hardcoded stuff for fullhd
            appWindow.MoveAndResize(new Windows.Graphics.RectInt32((1920 / 2) - width / 2, (1080 / 2) - height / 2, width, height));
#endif
```

**Reply by softlion** [reply]
WindowMapper does not exist. How do you get it.

**Reply by nor0x** [reply]
for >preview14 it's just `Mapper` i have updated my answer

**Reply by H4CK3ROFF** [reply]
How can I fix the size of the window? To prohibit the ability to scale the window. And remove the standard Title Bar or change its styles?

**Reply by PaulChek** [reply]
Thank you! It works perfectly)))

**Reply by LanceMcCarthy** [reply]
Here's the complete working version of this approach for RC1 (which is expected to keep working until GA):

```csharp
using Microsoft.Maui.LifecycleEvents;

#if WINDOWS
using Microsoft.UI;
using Microsoft.UI.Windowing;
using Windows.Graphics;
#endif

namespace MyApp.Maui
{
    public static class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                });

#if WINDOWS
            builder.ConfigureLifecycleEvents(events =>
            {
                events.AddWindows(wndLifeCycleBuilder =>
                {
                    wndLifeCycleBuilder.OnWindowCreated(window =>
                    {
                        IntPtr nativeWindowHandle = WinRT.Interop.WindowNative.GetWindowHandle(window);
                        WindowId win32WindowsId = Win32Interop.GetWindowIdFromWindow(nativeWindowHandle);
                        AppWindow winuiAppWindow = AppWindow.GetFromWindowId(win32WindowsId);

                        const int width = 1200;
                        const int height = 800;
                        winuiAppWindow.MoveAndResize(new RectInt32(1920 / 2 - width / 2, 1080 / 2 - height / 2, width, height));
                    });
                });
            });
#endif

            return builder.Build();
        }
    }
}
```

**Reply by ItsOswin** [reply]
So I tried this and saw theres a winuiAppWindow.Titlebar but it always stays null is there a way to get it ? I want to say ExtendsViewIntoTitlebar

**Reply by LanceMcCarthy** [reply]
@ItsOswin TitleBar features are still planned.
- You can see the WinApp SDK roadmap here => https://portal.productboard.com/winappsdk/1-windows-app-sdk/tabs/2-planned
- The planned Titlebar item is here => https://portal.productboard.com/winappsdk/1-windows-app-sdk/c/64-improved-default-and-custom-title-bar-via-windowing-apis

**Reply by ToolmakerSteve** [reply]
Updated for Maui GA:
```
public partial class App : Application
{
    const int WindowWidth = 400;
    const int WindowHeight = 300;
	public App()
	{
		InitializeComponent();

        Microsoft.Maui.Handlers.WindowHandler.Mapper.AppendToMapping(nameof(IWindow), (handler, view) =>
        {
#if WINDOWS
            var mauiWindow = handler.VirtualView;
            var nativeWindow = handler.PlatformView;
            nativeWindow.Activate();
            IntPtr windowHandle = WinRT.Interop.WindowNative.GetWindowHandle(nativeWindow);
            WindowId windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(windowHandle);
            AppWindow appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);
            appWindow.Resize(new SizeInt32(WindowWidth, WindowHeight));
#endif
        });

        MainPage = new MainPage();
    }
```

**Reply by sellorio2** [reply]
This fails to compile in my MAUI app that I created using the latest preview of Visual Studio 2 days ago (I'm pretty sure it's the bleeding edge still).

I don't know if it's relevant but I used the MAUI Blazor template.

```
error CS0246: The type or namespace name 'WindowId' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'AppWindow' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'SizeInt32' could not be found (are you missing a using directive or an assembly reference?)
```

**Reply by sabre230** [reply]
Adding this to the top of the file got it working for me. 

```
#if WINDOWS
using Microsoft.UI;
using Microsoft.UI.Windowing;
using Windows.Graphics;
#endif
```

**Comment by coderbasket** [comment]
There is a library here... with enums. css styling in c# and some cal().
each css enums has documentation and example along with various posible style values.
[https://github.com/coderbasket/StylerCode.Css](https://github.com/coderbasket/StylerCode.Css)

![alt text](https://raw.githubusercontent.com/coderbasket/StylerCode.Css/main/20220319_191849.jpg)

**Reply by nogginbox** [reply]
This looks like a generic CSS generation library and does not answer the question related to Blazor and WinUI.

**Reply by kla50** [reply]
I haven't worked with Blazor, but for WinUI I created the following extension method.  The WindowSettings type contains the position and size of the window and is part of my settings implementation. T is the root settings type that represents the entire appsettings.json file.  You'll need to replace this settings stuff with your settings implementation.

public static class WindowExtensions {
	public static void TrackWindowLocationAndSize<T>(this Window wnd_, WindowSettings wndSettings_) where T : BaseSettings {
		IntPtr hwnd = WinRT.Interop.WindowNative.GetWindowHandle(wnd_);
		PInvoke.User32.SetWindowPos(hwnd, PInvoke.User32.SpecialWindowHandles.HWND_TOP,
			(Int32)wndSettings_.X, (Int32)wndSettings_.Y, (Int32)wndSettings_.Width, (Int32)wndSettings_.Height,
			PInvoke.User32.SetWindowPosFlags.SWP_SHOWWINDOW);

		wnd_.Closed += (Object sender, WindowEventArgs args) => {
			IntPtr hwnd = WinRT.Interop.WindowNative.GetWindowHandle(wnd_);
			if(PInvoke.User32.GetWindowRect(hwnd, out PInvoke.RECT rect1)) {
				wndSettings_.X = rect1.left;
				wndSettings_.Y = rect1.top;
				wndSettings_.Width = rect1.right - rect1.left;
				wndSettings_.Height = rect1.bottom - rect1.top;
			}
			SettingsManager<T>.Current.Settings.Save();
		};
	}
}

Add a call this method to the application's window constructor:

public sealed partial class MainWindow : Window {
	internal MainWindowViewModel ViewModel { get; private set; }

	public MainWindow(MainWindowViewModel viewModel_, WindowSettings wndSettings_) {
		InitializeComponent();
		ViewModel = viewModel_;
		this.TrackWindowLocationAndSize<WinUI3Play01Settings>(wndSettings_);
	}
}

**Comment by ToolmakerSteve** [comment]
Updated for Maui GA:
```
public partial class App : Application
{
    const int WindowWidth = 400;
    const int WindowHeight = 300;
	public App()
	{
		InitializeComponent();

        Microsoft.Maui.Handlers.WindowHandler.Mapper.AppendToMapping(nameof(IWindow), (handler, view) =>
        {
#if WINDOWS
            var mauiWindow = handler.VirtualView;
            var nativeWindow = handler.PlatformView;
            nativeWindow.Activate();
            IntPtr windowHandle = WinRT.Interop.WindowNative.GetWindowHandle(nativeWindow);
            WindowId windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(windowHandle);
            AppWindow appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);
            appWindow.Resize(new SizeInt32(WindowWidth, WindowHeight));
#endif
        });

        MainPage = new MainPage();
    }
```

------------------
OR if want to base it on requested dimensions of MainPage, before appending handler could do:
```
        MainPage = new MainPage();
        var width = (int)MainPage.WidthRequest;
        var height = (int)MainPage.HeightRequest;
```

then use those dimensions (probably add some padding to get whole window size, because MainPage is client area).

**Reply by billn6** [reply]
Can you supply the full App.xaml.cs file, please?  I need the usings and related References to resolve the issues noted below.  I am using VS Community 17.3 preview 1.1.  My app is pretty close to a File | New .Net MAUI App, and works well enough aside from this.

![image](https://user-images.githubusercontent.com/6184923/175828514-a484343c-214a-490a-b0ad-a63a9b8730c0.png)

**Reply by LanceMcCarthy** [reply]
Just install WinUIEx package for the Windows and you can position and resize the windows in a 1-liner in MauiProgram.cs. No native code required.

step 1. Install WinUIEx NuGet package, but limit it only to the windows TFM:

YourPRoject.csproj

```xml
<Project Sdk="Microsoft.NET.Sdk">
...
	<ItemGroup>
		<PackageReference Include="WinUIEx" Version="1.5.0" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'" />
	</ItemGroup>
</Project>
```

Step 2. Update Maui.Program.cs to set your initial window size and position

In this example, I am using the `CenterOnScreen(width, height)` method because it automatically knows the user's dimensions!

```cs
using Microsoft.Maui.LifecycleEvents;

#if WINDOWS
using WinUIEx;
#endif

namespace YourApp;

public static class MauiProgram
{
  public static MauiApp CreateMauiApp()
  {
	  var builder = MauiApp.CreateBuilder();
	  builder
		  .UseMauiApp<App>()
		  .ConfigureFonts(fonts =>
		  {
			  fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
		  });
  
  #if WINDOWS
            builder.ConfigureLifecycleEvents(events =>
            {
                events.AddWindows(wndLifeCycleBuilder =>
                {
                    wndLifeCycleBuilder.OnWindowCreated(window =>
                    {
                        //Set size and center on screen using WinUIEx extension method
                        window.CenterOnScreen(1024,768); 
                    });
                });
            });
  #endif
  
            return builder.Build();
  }
}
```

**Reply by Ewerton** [reply]
> Can you supply the full App.xaml.cs file, please? I need the usings and related References to resolve the issues noted below. I am using VS Community 17.3 preview 1.1. My app is pretty close to a File | New .Net MAUI App, and works well enough aside from this.
> 
> ![image](https://user-images.githubusercontent.com/6184923/175828514-a484343c-214a-490a-b0ad-a63a9b8730c0.png)

@billn6 You could write this code inside the `YourProject\Platforms\Windows\App.xaml.cs` file. Then you dont need do worry about usings. This make more sense since you are changing just the Windows windows size.

**Reply by billn6** [reply]
@Ewerton, thanks, but that _was_ inside the App.xaml.cs.  I'm eager to hear back from @ToolmakerSteve, or see your working App.xaml.cs code if you have it available.

**Reply by billn6** [reply]
Thanks, @LanceMcCarthy. I might give that a try after I hear back from @ToolmakerSteve

**Reply by gfmoore** [reply]
That worked great, thanks. Just had to add a 
```
using Microsoft.UI.Windowing;
```

And because someone else will want to know I found that to position the window using
```
      appWindow.Move(new PointInt32(50,50));
```
seemed to work fine :)

**Reply by asmtzl** [reply]
ı did it, but how can ı lock the window. (ı dont want to user can change the size of window's height and width)

**Comment by pierre01** [comment]
It would be nice if someone publishes and make available the best practices for Desktop and Tablet Idioms.

**Reply by SimenCTan** [reply]
Is there a solution for mac?

**Reply by LanceMcCarthy** [reply]
This is what I use for MacCatalyst

```csharp
// In MauiProgram.cs inside `ConfigureLifecycleEvents`, directly after where you've already defined the WinUI window settings

#elif MACCATALYST
                
                events.AddiOS(wndLifeCycleBuilder =>
                {
                    wndLifeCycleBuilder.SceneWillConnect((scene, session, options) =>
                    {
                        if (scene is UIWindowScene { SizeRestrictions: { } } windowScene)
                        {
                            windowScene.SizeRestrictions.MaximumSize = new CGSize(1200, 900);
                            windowScene.SizeRestrictions.MinimumSize = new CGSize(600, 400);
                        }
                    });

                });
#endif
```

**Reply by MrZweistein** [reply]
From what package the events.AddiOS method comes from? When I use your code in my project I get an error that tells me that the keyword "events" is unknown "error CS0103".

Found the answer myself. The code block sits not in App.xaml.cs but in MauiProgram.cs !!

```
    public static MauiApp CreateMauiApp()
    {
        var builder = MauiApp.CreateBuilder();
        builder
            .UseMauiApp<App>()
            .ConfigureFonts(fonts =>
            {
                fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                fonts.AddFont("TWKEverett-Regular.otf", "EverettRegular");
                fonts.AddFont("TWKEverett-RegularItalic.otf", "EverettItalic");
                fonts.AddFont("TWKEverett-Bold.otf", "EverettBold");
            })
#if MACCATALYST
            .ConfigureLifecycleEvents(events =>
            {
                events.AddiOS(wndLifeCycleBuilder =>
                {
                    wndLifeCycleBuilder.SceneWillConnect((scene, session, options) =>
                    {
                        if (scene is UIKit.UIWindowScene { SizeRestrictions: { } } windowScene)
                        {
                            windowScene.SizeRestrictions.MaximumSize = new CoreGraphics.CGSize(1200, 900);
                            windowScene.SizeRestrictions.MinimumSize = new CoreGraphics.CGSize(600, 400);
                        }
                    });
                });
            })
#endif
            ;

        return builder.Build();
    }
```

**Reply by SimenCTan** [reply]
You have to refer the namespace UIKit

> From what package the events.AddiOS method comes from? When I use your code in my project I get an error that tells me that the keyword "events" is unknown "error CS0103".
> 
> Found the answer myself. The code block sits not in App.xaml.cs but in MauiProgram.cs !!
> 
> ```
>     public static MauiApp CreateMauiApp()
>     {
>         var builder = MauiApp.CreateBuilder();
>         builder
>             .UseMauiApp<App>()
>             .ConfigureFonts(fonts =>
>             {
>                 fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
>                 fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
>                 fonts.AddFont("TWKEverett-Regular.otf", "EverettRegular");
>                 fonts.AddFont("TWKEverett-RegularItalic.otf", "EverettItalic");
>                 fonts.AddFont("TWKEverett-Bold.otf", "EverettBold");
>             })
> #if MACCATALYST
>             .ConfigureLifecycleEvents(events =>
>             {
>                 events.AddiOS(wndLifeCycleBuilder =>
>                 {
>                     wndLifeCycleBuilder.SceneWillConnect((scene, session, options) =>
>                     {
>                         if (scene is UIKit.UIWindowScene { SizeRestrictions: { } } windowScene)
>                         {
>                             windowScene.SizeRestrictions.MaximumSize = new CoreGraphics.CGSize(1200, 900);
>                             windowScene.SizeRestrictions.MinimumSize = new CoreGraphics.CGSize(600, 400);
>                         }
>                     });
>                 });
>             })
> #endif
>             ;
> 
>         return builder.Build();
>     }
> ```

**Reply by MrZweistein** [reply]
Yeah, I already figured that out. My initial issue came from the fact that I thought the code that was provided was located in the App.xaml.cs. But in the MS docs I found that the code that was given by LanceMcCarthy belongs actually to MaiuProgram.cs.

**Reply by LanceMcCarthy** [reply]
I can see why that would be hard to determine because of the way GitHub groups the comments. Without the context of [my earlier reply](https://github.com/dotnet/maui/discussions/2370#discussioncomment-3027549), you wouldn't know it goes right after where you've defined your WinUI window settings. I've added a comment to the snippet in case it helps in the future.

**Reply by phollyer** [reply]
I've tried the above code for MacCatalyst but I'm getting the following error:

```
Error CS1061: 'MauiAppBuilder' does not contain a definition for 'ConfigureLifecycleEvents' and no accessible extension method 'ConfigureLifecycleEvents' accepting a first argument of type 'MauiAppBuilder' could be found (are you missing a using directive or an assembly reference?) (CS1061)
```

If I add the UIKit namespace VS reports that it can't be found.

Am I missing something?

**Reply by LanceMcCarthy** [reply]
@phollyer That error means you're missing a `using` statement. You can use VS IntelliSense (wait for lightbulb, then expand it... the tool should provide you with an option to add the missing using) or just copy from my demo here https://github.com/LanceMcCarthy/DevOpsExamples/blob/e66fa19aea5991158a2affbbc68efd3ea67b81e1/src/MAUI/MauiDemo/MauiProgram.cs#L2

**Reply by phollyer** [reply]
I was missing the following:

```

using Microsoft.Maui.LifecycleEvents;

#if WINDOWS
using WinUIEx;

#elif MACCATALYST
using CoreGraphics;
using UIKit;

#elif IOS
#elif ANDROID
#elif TIZEN

#endif
```

The app now compiles - but the maximum and minimum screen sizes aren't respected. I can still resize the window at will.

(Only tested for MacCatalyst)

**Reply by phollyer** [reply]
@LanceMcCarthy Definitely not getting any control over the window size for MacCatalyst.

https://github.com/dotnet/maui/assets/189019/795d596f-a4b8-4677-8cc6-c11fffe3a551

```C#
using Microsoft.Extensions.Logging;
using Microsoft.Maui.LifecycleEvents;

#if WINDOWS
using WinUIEx;

#elif MACCATALYST
using CoreGraphics;
using UIKit;

#elif IOS
#elif ANDROID
#elif TIZEN

#endif

namespace MauiPlayground;

public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			});

#if DEBUG
		builder.Logging.AddDebug();
#endif

        // Needs using Microsoft.Maui.LifecycleEvents for this extension method
        builder.ConfigureLifecycleEvents(events =>
        {
#if WINDOWS
            // using WinUIEx
            events.AddWindows(wndLifeCycleBuilder =>
            {
                wndLifeCycleBuilder.OnWindowCreated(window =>
                {
                    window.CenterOnScreen(1024,768); //uses WinUIEx
                });
            });

#elif MACCATALYST            
            // Using CoreGraphics and UIKit
            events.AddiOS(wndLifeCycleBuilder =>
            {
                wndLifeCycleBuilder.SceneWillConnect((scene, session, options) =>
                {
                    if (scene is UIWindowScene { SizeRestrictions: { } } windowScene)
                    {
                        windowScene.SizeRestrictions.MaximumSize = new CGSize(1200, 900);
                        windowScene.SizeRestrictions.MinimumSize = new CGSize(600, 400);
                    }
                });

            });
#endif
        });

        return builder.Build();
	}
}
```

Demo: https://github.com/phollyer/MauiPlayground/tree/FixedWindowSize

Is this a bug, or my code?

Thanks

**Reply by LanceMcCarthy** [reply]
@phollyer 
> Is this a bug, or my code?

I haven't been able to get much to work on maccatalyst in a while. I can't really recall when it did work, maybe MauiVersion 56 or 82.

I decided to stop developing it and wait for .NET8, there have just been too many regressions for Desktop in the past 6-9 months

**Reply by phollyer** [reply]
Ok thanks for the info.

**Comment by smourier** [comment]
FWIW, with latest MAUI (& WinAppSDK/WinUI3), it's much simpler, here is a full `App.xaml.cs `example:

```
using Microsoft.Maui.Platform;
using Windows.Graphics;

namespace MyApp
{
    public partial class App : MauiWinUIApplication
    {
        public App()
        {
            this.InitializeComponent();
            Microsoft.Maui.Handlers.WindowHandler.Mapper.AppendToMapping(nameof(IWindow), (handler, view) =>
            {
                handler.PlatformView.GetAppWindow().Resize(new SizeInt32(800, 400));
            });
        }

        protected override MauiApp CreateMauiApp() => MauiProgram.CreateMauiApp();
    }
}
```

It uses the `GetAppWindow` extension method from `Microsoft.Maui.Platform.WindowExtensions` class

**Comment by IT-VBFK** [comment]
I have a similar problem, but in my case my app window shrinks when running the app inside a remote session via a macBook Pro with retina display.

How can I achieve that my app window appears in same size in relation to screen width/height?
I know I can do some sort of calculation, but there must be a better method than that.

**Comment by sanepete** [comment]
Here's an update for MAUI Blazor hybrid with .Net 9.0

Edit the app.xaml.cs as follows:

```
namespace myawesomenamespace
{
  public partial class App : Application
  {
    public App()
    {
      InitializeComponent();
    }

    protected override Window CreateWindow(IActivationState? activationState)
    {
      Window? myawesomewindow = new Window(new MainPage()) { Title = "My Awesome Application" };
#if WINDOWS
      myawesomewindow .Width = 420;
      myawesomewindow .Height = 840;
#endif
      return myawesomewindow ;
    }
  }
}
```

**Comment by ITaluone** [comment]
Is there any way to disable window resize when moving to the top edge of the screen or double-click the title bar?
